### ISSUE TITLE
Question about what is Full Labeled Training and Datasets

### ISSUE THREAD

**Issue by sarmientoj24** [microsoft/SoftTeacher#34]
There required structure of the images is as follows:

```
# YOUR_DATA should be a directory contains coco dataset.
# For eg.:
# YOUR_DATA/
#  coco/
#     train2017/
#     val2017/
#     unlabeled2017/
#     annotations/
ln -s ${YOUR_DATA} data
bash tools/dataset/prepare_coco_data.sh conduct
```
My Questions are: 

1. If my understanding is correct, the `unlabeled2017` contains all the unlabeled images, right?

2. When you say X% labeled data (e.g. 5%, 10%, etc), does that take X% from the `train2017/` training data? What happens to the 100-X% of the data in the training data? Does it get added to the unlabeled pool for training?

3. When you say full-labeled training, does it mean it trains on all the data in `train2017/` (supervised) then use the `unlabeled2017/` data for unsupervised part of the semi-supervised learning? **Or** is it just supervised training on all training dataset? 

4. When using a custom dataset in COCO format, do I just follow the same instructions or do I need to change something more?

**Comment by MendelXu** [comment]
Q1: Yes
Q2: Yes. Yes.
Q3: Yes, the supervised baseline is trained on all labeled data (`train2017`) and the semi-supervised method is trained on all labeled data( `train2017` and `unlabeled2017`).
Q4: I think there is something you can check before your training: 1) Do you modify the annotation file path, image file prefix in the config file and replace them with your dataset configuration?; 2) Does your dataset shares the same categories with COCO? If not, add the following snippet to the config file.
```
data = dict(
     train=dict(
           ...
           classes=YOUR_CLASS_LIST
           ...
 ),
val=dict(
          ...
           classes=YOUR_CLASS_LIST 
          ...
),
test=dict(
           ...
           classes=YOUR_CLASS_LIST
          ... 
)
)
```

**Comment by sarmientoj24** [comment]
Thank you! Great answers and I appreciate the immediate response.
How do I do it if I want to use 100% of the labeled data (for supervised) and all the unlabeled data + labeled (for unsupervised) for semi-supervised training?

**Comment by MendelXu** [comment]
> Thank you! Great answers and I appreciate the immediate response. How do I do it if I want to use 100% of the labeled data (for supervised) and all the unlabeled data + labeled (for unsupervised) for semi-supervised training?

The following config should be OK.
```python
data = dict(
    train=dict(
        unsup=dict(
            _delete_=True,
            type="ConcatDataset",
            datasets=[
                dict(
                    typpe="CocoDataset",
                    ann_file="${data_root}annotations/instances_train2017.json",
                    img_prefix="${data_root}train2017",
                    pipeline="${train_unsup_pipeline}",
                ),
                dict(
                    typpe="CocoDataset",
                    ann_file="${data_root}annotations/instances_unlabeled2017.json",
                    img_prefix="${data_root}unlabeled2017",
                    pipeline="${train_unsup_pipeline}",
                ),
            ],
        ),
    )
)
```

**Comment by sarmientoj24** [comment]
Thanks @MendelXu 
Just to clarify. why does the second dictionary has an `ann_file` if those are unlabeled?
```
                dict(
                    typpe="CocoDataset",
                    ann_file="${data_root}annotations/instances_unlabeled2017.json",
                    img_prefix="${data_root}unlabeled2017",
                    pipeline="${unsup_pipeline}",
                    filter_empty_gt=False
                ),
```

Moreover, this means that the labeled samples should be inside `train2017` and the unlabeled inside `unlabeled2017`? Also, where should I insert the validation dataset here?

**Comment by MendelXu** [comment]
Q1: The json file contains some information like categories, image list, and so on.
Q2: No. You can store it anywhere you want and specify it in the config file 

There is a [tutorial](https://github.com/open-mmlab/mmdetection/blob/master/docs/tutorials/customize_dataset.md) provided `mmdetection` and you can refer to it about how to change the validation set, train set.

**Comment by sarmientoj24** [comment]
> Q1: The json file contains some information like categories, image list, and so on.
But what does it contain and how do I create that if those unlabeled images have no labels or bbox annotations at all?

> Q2: No. You can store it anywhere you want and specify it in the config file
I see. I just have to include the validation in this dictionary, right?

**Comment by MendelXu** [comment]
Yes
